mdagent/tools/base_tools/preprocess_tools/clean_tools.py

```diff
@@ -220,3 +221,157 @@ def _run(self, query: str) -> str:
     async def _arun(self, query: str) -> str:
         """Use the tool asynchronously."""
         raise NotImplementedError("custom_search does not support async")
+
+
+class CleaningToolFunctionInput(BaseModel):
+    """Input model for CleaningToolFunction"""
+
+    pdb_path: str = Field(..., description="Path to PDB or CIF file")
+    output_path: Optional[str] = Field(..., description="Path to the output file")
+    replace_nonstandard_residues: bool = Field(
+        True, description="Whether to replace nonstandard residues with standard ones. "
+    )
+    add_missing_atoms: bool = Field(
+        True,
+        description="Whether to add missing atoms to the file from the SEQRES records.",
+    )
+    remove_heterogens: bool = Field(
+        True, description="Whether to remove heterogens from the file."
+    )
+    remove_water: bool = Field(
+        True,
+        description="""Whether to remove water from the file.
+        remove_heterogens must be True.""",
+    )
+    add_hydrogens: bool = Field(
+        True, description="Whether to add hydrogens to the file."
+    )
+    add_hydrogens_ph: int = Field(7.0, description="pH at which hydrogens are added.")
+
+    @root_validator
+    def validate_query(cls, values) -> Dict:
+        """Check that the input is valid."""
+
+        return values
+
+
+class CleaningToolFunction(BaseTool):
+    name = "CleaningToolFunction"
+    description = """
+    This tool performs various cleaning operations on a PDB or CIF file.
+    Operations can include removing heterogens,
+    adding missing atoms and hydrogens,
+    replacing nonstandard residues, and/or removing water.
+
+    """
+    args_schema: Type[BaseModel] = CleaningToolFunctionInput
+
+    path_registry: Optional[PathRegistry]
+
+    def _run(self, **input_args) -> str:
+        """Use the tool with specified operations."""
+        try:
+            ### No idea why the input is a dictionary with the key "input_args"
+            # instead of the arguments themselves
+            print(input_args)
```

Comment:
also can we remove any unnecessary print statements that were used for debugging?

Reply:
fair!